great_expectations/profiler/rule/rule_state.py

```diff
@@ -115,14 +117,15 @@ def get_parameter_value(self, fully_qualified_parameter_name: str) -> Any:
         parameter_value: Optional[Any] = None
         try:
             for parameter_name_part in fully_qualified_parameter_as_list:
-                if parameter_container.descendants is None:
-                    parameter_value = parameter_container.parameters[
-                        parameter_name_part
-                    ]
-                else:
-                    parameter_container = parameter_container.descendants[
-                        parameter_name_part
-                    ]
+                if (
+                    parameter_container.parameters
+                    and parameter_name_part in parameter_container.parameters
+                ):
+                    return parameter_container.parameters[parameter_name_part]
+
+                parameter_container = parameter_container.descendants[
```

Comment:
Great refactor! To clarify:

`parameter_container.descendants` can be either `parameters` or `parameter_containers`?

Reply:
@talagluck In fact, they can only be of the `ParameterContainer` type -- that was the point of this typed object, to remove the confusion while overloading the standard Python `dict` type.

Reply:
Ah that makes sense - so a `parameter_container` could have both `parameters` and `descendants` the latter of which can only be `parameter_containers`. Given that these `descendants` can only be of one type, would it be worth re-considering the name to something like `descendant_parameter_containers`? Or does the typed object disambiguate enough? (I'm relatively new to typing in Python, so curious about conventions).

Reply:
@talagluck I believe that the latter in your observation is true.  The use of `descendants` is common.